cli/codegen/generate.go

```diff
@@ -0,0 +1,159 @@
+		PackageName: "repair",
+		FileName:    "cli/repair/lua_code_gen.go",
+		VariablesMap: map[string]string{
+			"reloadClusterwideConfigFuncBody": "cli/repair/lua/reload_clusterwide_config_func_body.lua",
+		},
+	},
+	generateLuaCodeOpts{
+		PackageName: "replicasets",
+		FileName:    "cli/replicasets/lua_code_gen.go",
+		VariablesMap: map[string]string{
+			"bootstrapVshardBody":                  "cli/replicasets/lua/bootstrap_vshard_body.lua",
+			"getClusterIsHealthyBody":              "cli/replicasets/lua/cluster_is_healthy_body.lua",
+			"editInstanceBody":                     "cli/replicasets/lua/edit_instance_body.lua",
+			"editReplicasetsBodyTemplate":          "cli/replicasets/lua/edit_replicasets_body_template.lua",
+			"formatTopologyReplicasetFuncTemplate": "cli/replicasets/lua/format_topology_replicaset_func_template.lua",
+			"getKnownRolesBody":                    "cli/replicasets/lua/known_roles_body.lua",
+			"getKnownVshardGroupsBody":             "cli/replicasets/lua/known_vshard_groups_body.lua",
+			"getMembershipInstancesBody":           "cli/replicasets/lua/membership_instances_body.lua",
+			"getTopologyReplicasetsBodyTemplate":   "cli/replicasets/lua/topology_replicasets_body_template.lua",
+			"probeInstancesBody":                   "cli/replicasets/lua/probe_instances_body.lua",
+		},
+	},
+}
+
+/* generateFileModeFile generates a file with map like this:
+
+var FileModes = map[string]int{
+	"filename": filemode,
+	...
+}
+*/
+func generateFileModeFile(path string, filename string) error {
+	f := jen.NewFile("static")
+	f.Comment("This file is generated! DO NOT EDIT\n")
+
+	fileModeMap, err := getFileModes(path)
+
+	if err != nil {
+		return err
+	}
+
+	f.Var().Id("FileModes").Op("=").Map(jen.String()).Int().Values(jen.DictFunc(func(d jen.Dict) {
+		for key, element := range fileModeMap {
+			d[jen.Lit(key)] = jen.Lit(element).Commentf("/* %#o */", element)
+		}
+	}))
+
+	f.Save(filename)
+
+	return nil
+}
+
+func getFileModes(root string) (map[string]int, error) {
+	fileModeMap := make(map[string]int)
+
+	err := filepath.Walk(root, func(filePath string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !fileInfo.IsDir() {
+			rel, err := filepath.Rel(root, filePath)
+
+			if err != nil {
+				return err
+			}
+
+			fileModeMap[rel] = int(fileInfo.Mode())
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return fileModeMap, nil
+}
+
+func generateLuaCodeVar() error {
+	for _, opts := range luaCodeFiles {
+		f := jen.NewFile(opts.PackageName)
+		f.Comment("This file is generated! DO NOT EDIT\n")
+
+		for key, val := range opts.VariablesMap {
+			content, err := ioutil.ReadFile(val)
+			if err != nil {
+				return err
+			}
+
+			f.Var().Id(key).Op("=").Lit(string(content))
+		}
+
+		f.Save(opts.FileName)
+	}
+
+	return nil
+}
+
+func main() {
+	err := generateFileModeFile(
+		"cli/create/templates/cartridge",
+		"cli/codegen/static/cartrdige_filemodes_gen.go",
```

Comment:
Let's rename it to `create_cartrdige_template_filemodes_gen`, we use "Cartridge" word anywhere in this code.